### ISSUE TITLE
Neighbors List link to legumemine: URL limit

### ISSUE THREAD

**Issue by sdash-github** [legumeinfo/lis_expression#15]
Apparently most browsers have 2083 characters limit for the URL.  If a gene model has 10 chars (approx.), the limit will be reached with a list of roughly 200 neighbors.  Any work around, now that neighbors can reach >2000?

**Comment by adf-ncgr** [comment]
yes, you will have to convert your request from a link-based GET to a form-based POST in order to support long lists. We did this for the GCV -> Intermine; here's the code:

`    <form id="legumemine-form" action='https://mines.legumeinfo.org/legumemine/bag.do' method="POST" target="_blank">
         <input type="hidden" name="type" value="Gene"/>
         <input type="hidden" name="text" value="{{geneListFormFormatted}}"/>
         <button onClick="document.getElementById('legumemine-form').submit();" type="submit">Create gene list in LegumeMine</button>
    </form>`
note that the {{geneListFormFormatted}} is interpolated into this template context and the
use of the button onClick to do the submit was a workaround for the straight HTML not seeming to function (ie submit the POST request) in the Angular context used by the GCV. I used target="_blank" here since there doesn't seem to be a way to let the user decide about whether to get a new window or not as you can do with right-click on hyperlinks in most browsers.

**Comment by sdash-github** [comment]
Will adopt this approach.

**Comment by sdash-github** [comment]
The 'newjson' lacks gene **name** of neighbors data, has only gene **uniquename**.  The link to mine must have gene **names**.
DO: Revise query and newjson to include gene <u>name</u> before making links for legume mine

**Comment by sdash-github** [comment]
The ongenome.genemodel table for later loaded genomes do not have chado genenames, have only chado.gene uniquenames in both columns.
So, reloading with one of the columns to gene names will help reformat the query. 
TO DO: (LATER) Discuss with Connor.
To be resolved later.

**Comment by sdash-github** [comment]
SQL changed in PHP function new_get_profile_neighbors_expression_json ($acc, $gene)  in lis_expression_helper_functions.inc to include genemodel_name
Results in the ajax string: ... "neighbor_uniquename":"Phvul.001G000800.v1.0","neighbor_name":"phavu.Phvul.001G000800" ... 

***CRITICAL*** 
When loaded the ongenome.genemodel.genemeodel_name field in most/all recent tables contains gene_uniquename except phavu.  So, it still won't work with this single change.  The genemodel table has to be reloaded/altered so that gene_name can be extracted and used for linking.
(Atleast I don't have to change the sql later.)

Will take up much later with Connor.
